duckduckgo/iOS
# Update Favicons to include 2 letters + colors (macOS parity)

Task/Issue URL: https://app.asana.com/0/1142021229838617/1205285857217353/f

**Description**:
Updates the favicon generator to support multiple characters.  It also makes sure we pass random colors for fake icon generation (which was not working before)



**Steps to test this PR**:
1. Add a few logins by hand
2. Add a few bookmarks by hand
3. Observe 2 letter favicons are created correctly
4. Smoke test the app looking for missing or broken favicons (Tabs, Bookmarks, Logins, Fireproof siles and unprotected sites)

<img width="565" alt="Screenshot 2023-08-21 at 3 12 37 PM" src="https://github.com/duckduckgo/iOS/assets/1156669/a030ae74-073f-440d-b271-1a98a6e45e2a">

<img width="565" alt="Screenshot 2023-08-21 at 3 12 50 PM" src="https://github.com/duckduckgo/iOS/assets/1156669/a2fb889b-c00a-4fc5-8300-ba1135681ce7">

<img width="565" alt="Screenshot 2023-08-21 at 3 13 06 PM" src="https://github.com/duckduckgo/iOS/assets/1156669/72b487c1-7d05-460c-912c-3f353060c4b7">

<img width="565" alt="Screenshot 2023-08-21 at 3 13 15 PM" src="https://github.com/duckduckgo/iOS/assets/1156669/2913d8b2-324e-4dac-8287-ec6539782f22">

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
